### PR TITLE
Prepare gpexpand Behave Test to use databases after ICW run.

### PR DESCRIPTION
**Pull request by kalensk**
For more comprehensive testing of our utilities such as gpexpand and gpgupgrade
we are using a dump after ICW runs. This dump.sql is stored on S3 for other jobs
to use.

However, gpexpand does not currently work with a ICW dump yet. So this commit
prepares for that step as we work to fix all the outstanding errors.

We have also made the following changes:

- Refactor and cleanup gpexpand behave tests

- Add gpexpand test that can be run locally

- Add new gpexpand test to verify redistribution

- Add initial gpexpand test using snowflake-simple-database until we fix
  gpexpand ICW dump errors

* Test Pipeline:
This PR is adding "icw-gporca-dump" resource after "icw_gporca_centos6" job and having gpexpand utilize the "icw-gporca-dump" resource.
https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-160560714-regression-database-part

Co-authored-by: Jamie McAtamney <jmcatamney@pivotal.io>
Co-authored-by: Jim Doty <jdoty@pivotal.io>
Co-authored-by: Kalen Krempely <kkrempely@pivotal.io>
Co-authored-by: Shoaib Lari <slari@pivotal.io>

### CODE REVIEW

**Comment by kalensk** [comment]
The initial commit has been split into two to separate getting the ICW cluster dump from the gpexpand tests. Also added the generated master pipeline, and fixed a bug where snowflake-simple-database-dump was added to gpexpand_1 and gpexpand_2 TINC tests.

File: concourse/tasks/ic_gpdb.yml [hunk 1]
```diff
@@ -5,7 +5,9 @@ inputs:
   - name: gpdb_src
   - name: bin_gpdb
 outputs:
+  - name: sqldump
```
Comment (kalensk) [inline concourse/tasks/ic_gpdb.yml]: Is this sqldump concourse "output" needed for the  icw-gporca-dump S3 "put" resource to succeeded? If not, should we consider removing it in favor of a simple "mkdir -p sqldump" in the dumpdb.bash script?

**Comment by jchampio** [comment]
> I am of the opinion that we should introduce the ICW dump additions with the PR that uses it.

@doty-pivotal Other upgrade work is currently blocked on having this dump available, so I was hoping we could submit that in parallel.

**Comment by doty-pivotal** [comment]
Cool - I am happy to submit as is if there is a consumer of the dump.

**Comment by doty-pivotal** [comment]
@kmacoskey - Pushed a commit with suggested changes, but since the YAML is so fun to read can you sanity check my work please?  After you review I will squash the `fixup!` before merging.

**Pull request merged**